aggregator/pom.xml

```diff
@@ -108,13 +111,68 @@
                 <executions>
                     <execution>
                         <id>main-${spark.version.classifier}</id>
-                        <phase>package</phase>
+                        <phase>compile</phase>
```

Comment:
The [maven-shade-plugin docs[(https://maven.apache.org/plugins/maven-shade-plugin/usage.html) state that the goals are bound to the package phase.  How does this not have issues with trying to shade before classes are compiled, and why is it necessary to change the phase to fix this?

Reply:
It is possible to change the phase in this module because it does not compile its own sources.  From my empiric observation, as a real inherent requirement the shade plugin only cares that the initial main artifact of  the project has been produced by the time it runs. This is achieved in this PR by setting 

`<rapids.default.jar.phase>initialize</rapids.default.jar.phase>` because it does not really matter when to produce an initially empty artifact

Changing is the phase is necessary to implement the memoization which happens in a different plugin, and there is no other way to establish the happens-before order because these are not executions of the same plugin  

The problem without memoization is described in #9691. The jar is updated unconditionally even if the output is identical to previous executions which triggers a full rebuild of 151 Scala files in `tests`  

My initial solution to that was to leave the aggregator module unchanged, introduce an intermediate module for memoization of the aggregator output and have `tests` depend on the intermediate instead of the original aggregator artifact.